src/quick-lint-js/parse.h

```diff
@@ -408,13 +414,15 @@ class parser {
   template <QLJS_PARSE_VISITOR Visitor>
   void visit_compound_assignment_expression(expression_ptr lhs,
                                             expression_ptr rhs, Visitor &v) {
-    this->visit_expression(lhs, v, variable_context::rhs);
+    bool used = false;
+    used = this->visit_expression(lhs, v, variable_context::rhs, true);
     this->visit_expression(rhs, v, variable_context::rhs);
-    this->maybe_visit_assignment(lhs, v);
+    this->maybe_visit_assignment(lhs, v, used);
```

Comment:
Along the same lines as my comment above, could we do this:
```suggestion
    if (used) {
      QLJS_ASSERT(lhs->kind() == expression_kind::variable);
      v.visit_variable_use_and_assignment(lhs->variable_identifier());
    } else {
      this->maybe_visit_assignment(lhs, v);
    }
```

Reply:
Done.